master/buildbot/steps/artifact.py

```diff
@@ -183,9 +187,15 @@ def start(self):
 
         if self.artifactBuildrequest:
             self.step_status.setText(["Artifact has been already generated."])
-            self.artifactPath = "%s_%s_%s" % (self.build.builder.config.builddir,
-                                              self.artifactBuildrequest['brid'],
-                                              FormatDatetime(self.artifactBuildrequest['submitted_at']))
+
+            if self.artifactBuildrequest["submitted_at"] > ARTIFACT_LOCATION_CHANGE_DATE:
+                self.artifactPath = "%s/%s_%s" % (self.build.builder.config.builddir,
```

Comment:
you are correct, though it would still mean recreating all artifacts (unnecessary extra time) and will probably break our artifact server (we don't have enough disk space to duplicate our current artifact load)